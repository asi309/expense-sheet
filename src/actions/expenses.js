import database from '../firebase/firebase';

export const addExpense = (expense) => ({
    type: 'ADD_EXPENSE',
    expense
});

export const startAddExpense = (
    { 
        description = '',
        note = '',
        amount = 0,
        createdAt = 0
    } = {}
) => {
    return (dispatch) => {
        const expense = { description, note, amount, createdAt }
        return database.ref('expenses').push(expense)
        .then((ref) => {
            dispatch(addExpense({
                id: ref.key,
                ...expense
            }));
        });
    };
};

export const removeExpense = (id) => ({
    type: 'REMOVE_EXPENSE',
    id
});

export const startRemoveExpense = (id) => {
    return (dispatch) => {
        database.ref(`expenses/${id}`).remove()
        .then(() => {
            dispatch(removeExpense(id));
        });
    };
};

export const editExpense = (id, updates) => ({
    type: "EDIT_EXPENSE",
    id,
    updates
});

export const startEditExpense = (id, updates) => {
    return (dispatch) => {
        database.ref(`expenses/${id}`).update(updates)
        .then(() => {
            dispatch(editExpense(id, updates));
        });
    };
};

// export const clearState = () => ({
//     type: "CLEAR_STATE"
// });

// export const startSetExpenses = () => {
//     return (dispatch) => {
//         database.ref('expenses').on('value', (snapshot) => {
//             dispatch(clearState());
//             const val = snapshot.val();
//             for(const key in val){
//                 dispatch(addExpense({
//                     id: key,
//                     ...val[key]
//                 }));
//             }
//         });
//     };
// }