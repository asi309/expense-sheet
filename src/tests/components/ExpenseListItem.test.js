import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses';

test('Should render ExpenseListItem with data', () => {
    const wrapper = shallow(<ExpenseListItem { ...expenses[0] } />);
    expect(wrapper).toMatchSnapshot();
});

test('Should render ExpenseListItem without data', () => {
    const wrapper = shallow(<ExpenseListItem />);
    expect(wrapper).toMatchSnapshot();
});