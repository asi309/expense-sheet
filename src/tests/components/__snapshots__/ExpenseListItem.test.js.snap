// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpenseListItem with data 1`] = `
<div>
  <h3>
    <Link
      to="/edit/1"
    >
       
      Gum
       
    </Link>
  </h3>
  <p>
    195
     - 
    10
  </p>
</div>
`;

exports[`Should render ExpenseListItem without data 1`] = `
<div>
  <h3>
    <Link
      to="/edit/undefined"
    >
       
       
    </Link>
  </h3>
  <p>
     - 
  </p>
</div>
`;
